stdsimd/arch/detect/linux/auxvec.rs

```diff
@@ -0,0 +1,290 @@
+//! Parses ELF auxiliary vectors.
+
+use core::mem;
+use _std::prelude::v1::*;
+use _std::fs::File;
+use _std::io::Read;
+
+#[cfg(not(all(target_os = "linux",
```

Comment:
Perhaps `cfg_if!` could be used to simplify this? That could be added as a dependency and it's available in libstd as well